### DISCUSSION TITLE
Invalidating a query key based on 2nd 3rd etc element in the query key array

### DISCUSSION THREAD

**Discussion by tomchambers2** [TanStack/query#7263] [Q&A]
Let's say you have a site with query keys like this:

```
["actions", 1234]
["action-log", 1234]
```

Is it possible to invalidate just `1234` so that both queries reload?

**Comment by TkDodo** [comment]
not like that, no. You would need to have objects, because they have no order:

```
[{ type: 'actions', id: 1234 }],
[{ type: 'actions-log', id: 1234 }]
```

then use:

```
queryClient.invalidateQueries({
  queryKey: [{ id: 1234 }]
})
```

or, you nest them like this:

```
['actions', 1234]
['actions', 1234, 'log']
```

**Comment by carlosvini** [comment]
I found this answer because I couldn't find the documented behavior of invalidateQueries when using object as part of the key.
I don't think I need to create a new topic since it's a discussion anyway, right? :) 

The page for https://tanstack.com/query/v5/docs/framework/react/guides/query-invalidation doesn't mention how object keys are filtered when only one of the properties is provided. Most of the examples are more about arrays (which I get is the most common use case).  Your example on this answer is the first result I could find that mentions this behavior (filtering by partial object).

Is there such documentation somewhere?

(I can't change the way my keys are structured. The project uses https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-react-query. I like the current behavior, I just wanted to confirm it was correct, and couldn't find the docs)

**Comment by aarjithn** [comment]
I had this same requirement, I thought queries do a match within whole of the array by default and then figured out its not.

I got it working like this though, preserving the existing query key structure.

``` ts
queryClient.invalidateQueries({
    predicate: (query) => query.queryKey.includes(1234)
})
```
  
  cc @TkDodo wonder if it makes sense adding this use case in the docs

**Reply by TkDodo** [reply]
> I thought queries do a match within whole of the array by default and then figured out its not

I don’t know what that means. please show a concrete example, how it behaves and how you would like it to behave instead.

**Reply by aarjithn** [reply]
I had a query defined with [“queryKey1”, queryKey2”] - these both are like tags and not hierarchical in our usecase. So it was not working when i do invalidateQueries([‘queryKey2’])

These are existing queries with queryKey2 in multiple positions so restructuring them is a big ask.

I think the current behaviour itself makes sense especially for hierarchical query keys, but this tag like structure could also be potential usecase, so using predicate to achieve this could be mentioned in docs?

**Reply by TkDodo** [reply]
`predicate` is a valid QueryFilter, which is described here: https://tanstack.com/query/latest/docs/framework/react/guides/filters#query-filters
